src/core/qgstemporalnavigationobject.cpp

```diff
@@ -148,6 +148,8 @@ void QgsTemporalNavigationObject::setTemporalExtents( const QgsDateTimeRange &te
     case NavigationOff:
       break;
   }
+
+  emit temporalExtentsChanged( mTemporalExtents );
```

Comment:
Can we also add a check in this function if the temporalExents are equal to mTemporalExtents then we should not update the mTemporalExtents or emit the temporalExentsChanged() signal.

Reply:
Yes, we can, and I did. Although throwing some extra signal should not be a problem is it?

Reply:
I think it is not a problem.